aspnetcore/fundamentals/middleware.md

```diff
@@ -161,33 +159,29 @@ ASP.NET ships with the following middleware components:
 
 ## Writing middleware
 
-The [CodeLabs middleware tutorial](https://github.com/Microsoft-Build-2016/CodeLabs-WebDev/tree/master/Module2-AspNetCore) provides a good introduction to writing middleware.
-
-For more complex request handling functionality, the ASP.NET team recommends implementing the middleware in its own class, and exposing an `IApplicationBuilder` extension method that can be called from the `Configure` method. The simple logging middleware shown in the previous example can be converted into a middleware class that takes in the next `RequestDelegate` in its constructor and supports an `Invoke` method as shown:
-
-RequestLoggerMiddleware.cs
+Middleware is generally encapsulated in a class and exposed with an extension method. Consider the following middleware, which sets the culture for the current request from the query string:
 
-[!code-csharp[Main](../fundamentals/middleware/sample/src/MiddlewareSample/RequestLoggerMiddleware.cs?highlight=12,18)]
+[!code-csharp[Main](middleware/sample/Culture/StartupCulture.cs?name=snippet1)]
 
+Note: The sample code above is used to demonstrate creating a middleware component. See [
```

Comment:
For sure we don't want people to copy/paste and write an app based on this (rather than using the localization middleware). I can make this a note if you think that's would expose it better. People often skip note dialogues.